app/common/utils/logger/logger.go

```diff
@@ -31,6 +34,13 @@ func createLogger(env string) *zap.Logger {
 	return newLogger
 
 }
+
+func (l Log) SetRequestIDFromContext(ctx context.Context) logHelper.Logger {
```

Comment:
Por ser um método `value receiver`, acredito que essa mutação não irá funcionar como esperado. Chegou a verificar se depois que o método é chamado com um contexto que contenha o valor, o request id sai nos logs seguintes?

Reply:
Sim, ele sai nos valores corretamente:
2022-02-01T16:41:39.947-0300    INFO    received request in url: /login {"environment": "development", "requestID": "45ee7920-6026-48d4-8ca4-7e4b3faa28a2", "operation:": "request"}

Um trecho de exemplo, onde loguei o requestID na rota que eu implementei.

Reply:
Obseve que nos casos onde o método foi invocado, houve uma nova declaração para armazenar o valor de retorno, por exemplo:
`log := c.log.SetRequestIDFromContext(r.Context())`

Se a operação for apenas `c.log.SetRequestIDFromContext(r.Context())` esse método não irá funcionar como esperado.

Reply:
Sim, porque ela tem como retorno um novo logger. Mas isso foi feito como era esperado: para retornar um novo logger, que deverá ser usado para os logs daquela função.

Qual seria a sugestão de melhoria nesse caso?